plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -4133,3 +4133,48 @@ function wc_set_hooked_blocks_version_on_theme_switch( $old_name, $old_theme ) {
 		add_option( $option_name, WC()->version );
 	}
 }
+
+/**
+ * Add aria-label to pagination numbers.
+ *
+ * @param string $html HTML output.
+ * @param array  $args An array of arguments. See paginate_links()
+ *                     for information on accepted arguments.
+ *
+ * @return string
+ */
+function wc_add_aria_label_to_pagination_numbers( $html, $args ) {
+	$p         = new WP_HTML_Tag_Processor( $html );
+	$n         = 1;
+	$page_text = __( 'Page', 'woocommerce' );
+
+	while ( $p->next_tag( array( 'class_name' => 'page-numbers' ) ) ) {
+		if (
+			$p->has_class( 'prev' ) ||
+			$p->has_class( 'next' ) ||
+			( 'SPAN' !== $p->get_tag() && 'A' !== $p->get_tag() ) ) {
+			continue;
+		}
+
+		if ( $p->has_class( 'current' ) ) {
+			$n = $args['current'];
+		}
+
+		if ( $p->has_class( 'dots' ) ) {
+			if ( $args['current'] - $args['mid_size'] > $n ) {
+				$n = $args['current'] - $args['mid_size'] - 1;
+			} else {
+				$n = $args['total'] - $args['end_size'];
+			}
+			++$n;
+			continue;
+		}
+
+		$p->set_attribute( 'aria-label', $page_text . ' ' . number_format_i18n( $n ) );
+		++$n;
+	}
```

Comment:
I tested the PR and it works fine. I'm wondering though if we cannot simplify this logic a bit. Rather than excluding elements like ellipsis and prev/next page, what if we simply look for `$p->has_class( 'page-number' )` and assign pages to them using [`get_modifiable_text()`](https://developer.wordpress.org/reference/classes/wp_html_tag_processor/get_modifiable_text/).

I imagined something like:

```suggestion
		if (
			$p->has_class( 'page-number' )
			    $n = $p->get_modifiable_text();
           		$p->set_attribute( 'aria-label', $page_text . ' ' . number_format_i18n( $n ) );
		}
	}
```

What do you think about such approach? Is there a need to calculate `$n` manually?

Reply:
@kmanijak Thanks for your feedback!

The reason why I didn't use the `get_modifiable_text()` method is that users can pass the `before_page_number` and `after_page_number` arguments to modify the page number text, but we don't want them to be part of the `aria-label` value.

By calculating `$n`, we ensure that only the page number is passed to the `aria-label` attribute.

Reply:
Makes sense, thanks for explanation!